csm/conf/cleanup.py

```diff
@@ -59,3 +63,51 @@ def files_directory_cleanup(self):
     def web_env_file_cleanup(self):
        Log.info(f"Replacing {const.CSM_WEB_DIST_ENV_FILE_PATH}_tmpl {const.CSM_WEB_DIST_ENV_FILE_PATH}")
        Setup._run_cmd(f"cp -f {const.CSM_WEB_DIST_ENV_FILE_PATH}_tmpl {const.CSM_WEB_DIST_ENV_FILE_PATH}")
+
+    def cleanup_ldap_config(self):
+        #Delete cortxuser schema
+        filelist = glob.glob('/etc/openldap/slapd.d/cn=config/cn=schema/*cortxuser.ldif')
+        for schemafile in filelist:
+            try:
+                os.remove(schemafile)
+            except Exception as e:
+                Log.error(f"Failed to delete cortxuser schema: {e}")
+                raise CsmSetupError("Failed to delete cortxuser schema.")
+        #Delete permission attributes
+        try:
+            self._search_delete_permission_attr("olcDatabase={2}mdb,cn=config", "olcAccess")
```

Comment:
Are you supposed to remove permissions first and then the schema?

Reply:
No. Sequence does not matter.